test/controllers/admin/comfy/external_api_clients_controller_test.rb

```diff
@@ -38,27 +75,402 @@ class Comfy::Admin::ExternalApiClientsControllerTest < ActionDispatch::Integrati
     assert_response :success
   end
 
-  test "should get new" do
-    skip
+  test "#new: denies if user not signed in" do
+    get new_api_namespace_external_api_client_path(api_namespace_id: @api_namespace.id)
+    expected_message = "You need to sign in or sign up before continuing."
+
+    assert_response :redirect
+    assert_match expected_message, flash[:alert]
+  end
+
+  test "#new: denies if user not permissioned to manage api" do
+    @user.update(can_manage_api: false)
+    sign_in(@user)
+
+    get new_api_namespace_external_api_client_path(api_namespace_id: @api_namespace.id)
+    expected_message = "You do not have the permission to do that. Only users who can_manage_api are allowed to perform that action."
+
+    assert_response :redirect
+    assert_match expected_message, flash[:alert]
+  end
+
+  test "#new: allows if permissioned user is signed in" do
+    @user.update(can_manage_api: true)
+    sign_in(@user)
+
+    get new_api_namespace_external_api_client_path(api_namespace_id: @api_namespace.id)
+
+    assert_response :success
+  end
+
+  test "#create: should allow when permissioned user is signed-in" do
+    payload = {
+      external_api_client: {
```

Comment:
@alis-khadka can you add some assertions to this test that validates passed parameters are saved correctly on the DB (eg: metadata hash is saved the with the same keys/values)